### ISSUE TITLE
-Wunsafe-buffer-usage is useless

### ISSUE THREAD

**Issue by lucas-mior** [llvm/llvm-project#65071]
Hello, I strongly suggest to remove this warning, or at least remove from the `-Weverything` option.

Before clang 16.0 I was able to compile all my projects using `-Weveything` without problems,
fixing my code and paying attention to the (useful) warnings that this option gave.
However, it seems that the `-Wunsafe-buffer-usage` was added in version 16 (my system updated to it recently)
and then compiling any project is impossible without getting a huge list of useless warnings about *any* pointer
being used, even in main:

```
main.c:25:26: warning: 'argv' is an unsafe pointer used for buffer
access [-Wunsafe-buffer-usage]
int main(int argc, char *argv[]) {
                   ~~~~~~^~~~~~
main.c:35:21: note: used in buffer access here
```

One can't possibly reason about his code using this warning.
Currently I disable it manually with `-Wno-unsafe-buffer-usage`, but it is an ugly solution
and I wonder (please correct me) if this causes to disable other warnings that could be useful
and got mixed up with this kind of warning.

Thanks.

**Comment by shafik** [comment]
Note as documented: https://clang.llvm.org/docs/UsersManual.html#diagnostics-enable-everything

> Since [-Weverything](https://clang.llvm.org/docs/UsersManual.html#cmdoption-Weverything) enables every diagnostic, we generally don’t recommend using it. -Wall -Wextra are a better choice for most projects. Using [-Weverything](https://clang.llvm.org/docs/UsersManual.html#cmdoption-Weverything) means that updating your compiler is more difficult because you’re exposed to experimental diagnostics which might be of lower quality than the default ones. If you do use [-Weverything](https://clang.llvm.org/docs/UsersManual.html#cmdoption-Weverything) then we advise that you address all new compiler diagnostics as they get added to Clang, either by fixing everything they find or explicitly disabling that diagnostic with its corresponding Wno- option.

**Comment by lucas-mior** [comment]
Thanks for the answer.
Could you please explain how would I adress new diagnostics by "fixing everything they find".
For instance, how could I make clang not complain about using `argv` (without disabling this warning explicitly)?
